content/k8s/fluxcd/automation/_index.md

```diff
@@ -0,0 +1,148 @@
+---
+content_type: project
+flavours:
+- none
+from_repo: k8s/manual-app-deployment/project-overview
+prerequisites:
+  hard:
+  - k8s/manual-app-deployment/project-overview
+  soft:
+  - k8s/fluxcd/postgresql-buttons
+ready: true
+submission_type: continue_repo
+tags:
+- kubernetes
+- fluxcd
+title: Bootstrap flux
+
+# Automation
+
+Although we are now mostly automated we still need to push code then edit the values file to reflect the new docker tag to rollout new code, how about we automate this step
+
+Let's get flux to start staring at the docker-registry by using a imagerepository
+
+There are 4 things that make the automations work:
+- ImageRepository: this is used to watch your harbor registry for new tags
+- ImagePolicy: The decision when a new tag is discovered whether to roll them out
+- ImageUpdateAutomation: The action of where to make the update in your gitrepository for the tag it discovered
+- # {"$imagepolicy": "flux-system:nginx:tag"} : in the *.yaml file where you want to update the tag you need to tell flux where to make the commit.
+
+
+create the file
+
+Copy the secret from buttons namespace to flux-system so that it can log in to image registry
+
+```
+kubectl get secret your-regcred -n buttons -o yaml | sed 's/namespace: buttons/namespace: flux-system/' | kubectl apply -n flux-system -f -
+```
+
+```
+# apps/automation.yaml
+---
+apiVersion: image.toolkit.fluxcd.io/v1beta2
+kind: ImageRepository
+metadata:
+  name: nginx
+  namespace: flux-system
+spec:
+  image: <my.domain.com>/application/nginx
+  interval: 1m
+  secretRef:
+    name: your-regcred
+```
+
+Now check if the imagerepository is loaded
+
+```
+kubectl -n buttons get ImageRepository
+```
+
+```
+# apps/automation.yaml
```

Comment:
Since we are going to use the same file, can you clarify to keep appending content to the file?